packages/templating/dynamic_tests.js

```diff
@@ -86,6 +86,59 @@ Tinytest.add(
 );
 
 Tinytest.add(
+  "spacebars - ui-dynamic-template - render template dynamically with contentBlock", function (test, expect) {
+    var tmpl = Template.ui_dynamic_test_contentblock;
+
+    var nameVar = new ReactiveVar;
+    var dataVar = new ReactiveVar;
+    tmpl.helpers({
+      templateName: function () {
+        return nameVar.get();
+      },
+      templateData: function () {
+        return dataVar.get();
+      }
+    });
+
+    // No template chosen
+    var div = renderToDiv(tmpl);
+    test.equal(canonicalizeHtml(div.innerHTML), "");
+
+    // Choose the "ui-dynamic-test-sub" template, with no data context
+    // passed in.
+    nameVar.set("ui_dynamic_test_sub_contentblock");
+    Tracker.flush();
```

Comment:
In tests, we use `Tracker.flush({_throwFirstError: true})` so that if there's an error here it fails rather than prints an error asynchronously to the console.